src/core/session.cpp

```diff
@@ -60,6 +77,31 @@ class standard_logger_impl : public logger_impl
         return lastQuery_;
     }
 
+    virtual std::string get_last_query_with_context() const
+    {
+        if (queryParams_.empty()) {
+            return get_last_query();
+        }
+
+        std::string query = get_last_query();
+
+        query += " with ";
+
+        for (std::size_t i = 0; i < queryParams_.size(); ++i)
```

Comment:
I'd suggest still using the ranged for loop, it's just much more clear:

```suggestion
        bool first = true;
        for (const auto& param: queryParams_)
```

and then in the loop body

```cpp
if (first)
    first = false;
else
    query += ", ";
```